block/submit.go

```diff
@@ -61,50 +61,65 @@ func (m *Manager) handleSubmissionTrigger(ctx context.Context) {
 }
 
 func (m *Manager) submitNextBatch() (uint64, error) {
-	// Get the batch start and end height
-	startHeight := m.syncTarget.Load() + 1
-	endHeight := uint64(m.store.Height())
+	//check if pending batch exists.
+	if m.pendingBatch != nil {
```

Comment:
Possibly be cleaner to extract all the batch fetching/creation to an external method as this method currently does 3 things:
1. creates batch (or gets pending one)
2. submits to DA
3. submit to Settlement.